src/main/java/org/clas/detectors/AHDCmonitor.java

```diff
@@ -128,11 +208,29 @@ public void processEvent(DataEvent event) {
     @Override
     public void analysisUpdate() {
         if(this.getNumberOfEvents()>0) {
-            H2F raw = this.getDataGroup().getItem(0,0,0).getH2F("rawADC");
-            for(int loop = 0; loop < raw.getDataBufferSize(); loop++){
-                this.getDataGroup().getItem(0,0,0).getH2F("occADC").setDataBufferBin(loop,100*raw.getDataBufferBin(loop)/this.getNumberOfEvents());
+	    H2F hist2d_raw_occupancy = this.getDataGroup().getItem(0,0,0).getH2F("raw_occupancy");
+            for(int ibin = 0; ibin < hist2d_raw_occupancy.getDataBufferSize(); ibin++){
+            	float raw_occupancy = hist2d_raw_occupancy.getDataBufferBin(ibin);
+		if (raw_occupancy != 0) {
+		    // recover all raw values
+		    float raw_adcMax = this.getDataGroup().getItem(0,0,0).getH2F("raw_adcMax").getDataBufferBin(ibin);
+		    float raw_integral = this.getDataGroup().getItem(0,0,0).getH2F("raw_integral").getDataBufferBin(ibin);
+		    float raw_timeMax = this.getDataGroup().getItem(0,0,0).getH2F("raw_timeMax").getDataBufferBin(ibin);
+		    float raw_leadingEdgeTime = this.getDataGroup().getItem(0,0,0).getH2F("raw_leadingEdgeTime").getDataBufferBin(ibin);
+		    float raw_timeOverThreshold = this.getDataGroup().getItem(0,0,0).getH2F("raw_timeOverThreshold").getDataBufferBin(ibin);
+		    float raw_constantFractionTime = this.getDataGroup().getItem(0,0,0).getH2F("raw_constantFractionTime").getDataBufferBin(ibin);
+		    // renormalise them
+		    this.getDataGroup().getItem(0,0,0).getH2F("occupancy").setDataBufferBin(ibin, raw_occupancy/1.0);
```

Comment:
For this to be the true occupancy, raw_occupancy should be divided by the number of events (and multiplied by 100 if using % as units)